webui/react/src/components/NotebookModal.tsx

```diff
@@ -0,0 +1,247 @@
+        <div className={css.note}
+        >
+          <Link external path="/docs/reference/command-notebook-config.html">
+          Read about notebook settings
+          </Link>
+        </div><React.Suspense
+          fallback={<div className={css.loading}><Spinner className="minHeight" /></div>}>
+          <Item
+            name="config"
+            rules={[ { message: 'Invalid YAML', required: true }, () => ({
+              validator(_, value) {
+                try {
+                  yaml.load(value);
+                  return Promise.resolve();
+                } catch(err) {
+                  return Promise.reject(new Error('Invalid YAML'));
+                }
+              },
+            }) ]}>
+            <MonacoEditor
+              height={400}
+              language="yaml"
+              options={{
+                minimap: { enabled: false },
+                scrollBeyondLastLine: false,
+                selectOnLineNumbers: true,
+              }} />
+          </Item>
+        </React.Suspense>
+      </Form> :
+      <Form
+        form={form}
+        initialValues={storage.getWithDefault(STORAGE_KEY, {
+          slots:1,
+          template: 'default',
+          type: undefined,
+        })}
+        labelCol={{ span:8 }}
+        onValuesChange={storeConfig}>
+        <Item label="Notebook Template" name="template">
+          <Select>
+            <Option key="default" value="default">Default Task Template</Option>
+            {templates.map(temp =>
+              <Option key={temp.name} value={temp.name}>{temp.name}</Option>)}
+          </Select>
+        </Item>
+        <Item label="Name" name="name">
+          <Input placeholder="Name" />
+        </Item>
+        <Item
+          label="Resource Pool"
+          name="pool"
+          rules={[ { message: 'Select a resource pool', required: true } ]}>
+          <Select
+            placeholder="Select a resource pool"
+            onChange={handleResourcePoolUpdate}>
+            {resourcePools.map(pool =>
+              <Option key={pool.name} value={pool.name}>{pool.name}</Option>)}
+          </Select>
+        </Item>
+        <Row justify="end">
+          <Col span={8}>
+            <Item
+              label="Type"
+              labelCol={{ span:8 }}
+              name="type"
+              rules={[ { message: 'Select a resource type', required: true } ]}>
+              <RadioGroup
+                options={resourceTypeOptions}
+                onChange={handleTypeUpdate} />
+            </Item>
+          </Col>
+          <Col span={11}>
+            { resourceType === 'GPU'?
+              <Item
+                label="Number of Slots"
+                labelCol={{ span:14 }}
+                name="slots"
+                rules={[ { message: 'Please choose a number of slots', required: true } ]}>
+                <InputNumber min={1} />
+              </Item> : null
+            }
+          </Col>
+        </Row>
```

Comment:
**chore:** let me know if you want to discuss this one in more detail:

![2021-05-20 at 1 28 PM](https://user-images.githubusercontent.com/220971/119038058-ee923600-b96f-11eb-89ae-e1097b3fc308.png)

Reply:
@hkang1 Resoruce pools should actually be optional. Sorry for the recent change. We used to say we can label resource pools to be default CPU/GPU pools. But we recently decided that there might be many different types of resource pools, not limited to CPU and GPU. It might not be ideal to choose any resource pool to be the default so I think it's okay to allow it to be empty, which is the same behavior as the CLI.